components/brave_wallet/browser/brave_wallet_service.cc

```diff
@@ -289,3 +291,40 @@ void BraveWalletService::OnExtensionLoaded(
     const extensions::Extension* extension) {
   RemoveUnusedWeb3ProviderContentScripts();
 }
+
+bool BraveWalletService::IsCryptoWalletsSetup() const {
+  PrefService* prefs = user_prefs::UserPrefs::Get(context_);
+  return prefs->HasPrefPath(kBraveWalletAES256GCMSivNonce) &&
```

Comment:
I'm not familiar with `HasPrefPath`, will this just check if the preference exists at all on the profile, or if it is non-empty? I think it's the former it should be an non-empty check.

Reply:
I think it's correct:
```
  // Returns true if a value has been set for the specified path.
  // NOTE: this is NOT the same as FindPreference. In particular
  // FindPreference returns whether RegisterXXX has been invoked, where as
  // this checks if a value exists for the path.
  bool HasPrefPath(const std::string& path) const;
```